### PR TITLE
Notification payload description string escaping

### PR DESCRIPTION
**Pull request by borissundic**
Hello,

I had a problem getting notifications when using [[DESCRIPTIONS]] in payload for Slack. Problem was in descriptions produced by .AddSqlServer health check, because they contained special characters like quotes and new lines when database goes offline. I made a fix escaping descriptions, so they don't break json formating.

Thank you.

### CODE REVIEW

**Comment by CarlosLanderas** [comment]
Thanks for reporting this.  Could you please provide a sample crashing payload?

**Comment by borissundic** [comment]
Sure.

This one is failing:
"{\"text\":\"HealthCheck: CrudServis failing - 'There are at least 1 HealthChecks failing.' ☢ Cannot open database \"Podaci\" requested by the login. The login failed.\r\nLogin failed for user 'D-3\\Korisnik'.\",\"channel\":\"#programeri\",\"link_names\": 1,\"username\":\"Boris\",\"icon_emoji\":\":monkey_face:\"}"

and this one, fixed, is passing:
"{\"text\":\"HealthCheck: CrudServis failing - 'There are at least 1 HealthChecks failing.' ☢ Cannot open database \\\"Podaci\\\" requested by the login. The login failed.\r\nLogin failed for user 'D-3\\\\Korisnik'.\",\"channel\":\"#programeri\",\"link_names\": 1,\"username\":\"Boris\",\"icon_emoji\":\":monkey_face:\"}"

**Comment by unaizorrilla** [comment]
@CarlosLanderas  are you following this?

**Comment by CarlosLanderas** [comment]
@borissundic  thanks for the contribution, please could you address the code review?

**Comment by borissundic** [comment]
@CarlosLanderas I think I don't have enough rights for that

**Comment by unaizorrilla** [comment]
Hi @borissundic 

Can you address the feedback from @CarlosLanderas ?

File: src/HealthChecks.UI/Core/Notifications/WebHookFailureNotifier.cs [hunk 1]
```diff
@@ -59,6 +61,16 @@ await SaveNotification(new HealthCheckFailureNotification()
                 _logger.LogInformation("Notification is sent on same window time.");
             }
         }
+        private string StringEscape(string str)
+        {
+            StringBuilder s = new StringBuilder(str);
```
[the remark this answers is not in the document]
Comment (borissundic) [reply src/HealthChecks.UI/Core/Notifications/WebHookFailureNotifier.cs]: You're right, it's much simpler with JavaScriptEncoder

**Comment by borissundic** [comment]
@CarlosLanderas Thanks for review. I made changes.

**Comment by insideou7** [comment]
This should be encoding all parameters.

IE
`
            payload = payload.Replace(Keys.LIVENESS_BOOKMARK, HttpUtility.JavaScriptStringEncode(name))
                .Replace(Keys.FAILURE_BOOKMARK, HttpUtility.JavaScriptStringEncode(failure))
                .Replace(Keys.DESCRIPTIONS_BOOKMARK, HttpUtility.JavaScriptStringEncode(description));
`

**Comment by enthiric** [comment]
Hello, I was wondering what is still blocking this improvement as I'm currently experiencing the same issue with our slack webhook. At the moment any failing health checks coming in from our database that result in a notification are basically gone.

**Comment by CarlosLanderas** [comment]
@enthiric thanks for reporting. @borissundic can you address comments in this PR ?

**Comment by CarlosLanderas** [comment]
@enthiric could you provide some payload samples please?.

**Comment by enthiric** [comment]
@CarlosLanderas sure this is from the MongoDB health check (package AspNetCore.HealthChecks.MongoDb) using the following payload template:
`[HEALTH CHECK] [DEVELOPMENT] [[LIVENESS]] is FAILING - Reason: [[DESCRIPTIONS]]`

`{"text":"[HEALTH CHECK] [DEVELOPMENT] basic is FAILING - Reason: A timeout occured after 30000ms selecting a server using CompositeServerSelector{ Selectors = MongoDB.Driver.MongoClient+AreSessionsSupportedServerSelector, LatencyLimitingServerSelector{ AllowedLatencyRange = 00:00:00.0150000 } }. Client view of cluster state is { ClusterId : "1", ConnectionMode : "Automatic", Type : "Standalone", State : "Disconnected", Servers : [{ ServerId: "{ ClusterId : 1, EndPoint : "Unspecified/localhost:27017" }", EndPoint: "Unspecified/localhost:27017", State: "Disconnected", Type: "Unknown", HeartbeatException: "MongoDB.Driver.MongoConnectionException: An exception occurred while opening a connection to the server.
 ---> System.Net.Internals.SocketExceptionFactory+ExtendedSocketException (61): Connection refused [::1]:27017
   at System.Net.Sockets.Socket.EndConnect(IAsyncResult asyncResult)
   at System.Net.Sockets.Socket.<>c.<ConnectAsync>b__274_0(IAsyncResult iar)
--- End of stack trace from previous location where exception was thrown ---
   at MongoDB.Driver.Core.Connections.TcpStreamFactory.ConnectAsync(Socket socket, EndPoint endPoint, CancellationToken cancellationToken)
   at MongoDB.Driver.Core.Connections.TcpStreamFactory.CreateStreamAsync(EndPoint endPoint, CancellationToken cancellationToken)
   at MongoDB.Driver.Core.Connections.BinaryConnection.OpenHelperAsync(CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at MongoDB.Driver.Core.Connections.BinaryConnection.OpenHelperAsync(CancellationToken cancellationToken)
   at MongoDB.Driver.Core.Servers.ServerMonitor.HeartbeatAsync(CancellationToken cancellationToken)", LastUpdateTimestamp: "2020-04-17T11:06:45.7594240Z" }] }","channel":"#example","link_names": 1,"username":"Notification bot"}
`

**Comment by CarlosLanderas** [comment]
#499

**Pull request closed without merging**